docs: Updated onboarding documentation

MoritzThomasHuebner opened the pull request:
## Summary

Updates the onboarding documentation for setting up magma with Ubuntu 22.04

- Added instructions about how to install Vagrant to avoid issues
- `python-ssl` -> `python3-ssl`
- Updated python version to be installed
- Corrected the `fabric` version since `fabric3` can not actually be used
- Added instructions to add `VAGRANT_DEFAULT_PROVIDER` environment variable
- Added information that browser certificate ownership needs to be changed to the user as they are created with root ownership
- Added more detailed instructions as to how to check "Enable all networks" in the NMM, as the option is somewhat hidden.

## Test Plan

Run unit tests

## Additional Information

- [ ] This change is backwards-breaking

MoritzThomasHuebner committed:
fix(docs): Updated onboarding documentation

Updates the onboarding documentation for setting up magma with Ubuntu 22.04

- Added instructions about how to install Vagrant to avoid issues
- `python-ssl` -> `python3-ssl`
- Updated python version to be installed
- Corrected the `fabric` version since `fabric3` can not actually be used
- Added instructions to add `VAGRANT_DEFAULT_PROVIDER` environment variable
- Added information that browser certificate ownership needs to be changed to the user as they are created with root ownership
- Added more detailed instructions as to how to check "Enable all networks" in the NMM, as the option is somewhat hidden.

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

voisey commented on docs/readmes/basics/prerequisites.md:
````diff
@@ -129,6 +131,9 @@ Development can occur from multiple OS's, where **macOS** and **Ubuntu** are **e
       pyenv global 3.7.3
       ```
 
+        **Note**: Python 3.7.3 cannot be installed on Ubuntu 22.04 this way. Use Python 3.10.4 instead.
````
Could installing a more up to date Python version potentially cause issues in some places? Perhaps it should be recommended that 3.7.3 is installed via a different method instead of using a different version?

MoritzThomasHuebner replied:
After doing some research, this seems to be an issue with pyenv for different system configurations: https://github.com/pyenv/pyenv/issues/2046

I changed the note to provide a workaround for 3.7.3 should the installation fail.

voisey commented on docs/readmes/basics/quick_start_guide.md:
```diff
@@ -133,15 +133,26 @@ Creating orc8r_controller_1       ... done
 The Orchestrator application containers will bootstrap certificates on startup
 which are cached for future runs. Watch the directory `magma/.cache/test_certs`
-for a file `admin_operator.pfx` to show up (this may take a minute or 2), then:
+for a file `admin_operator.pfx` to show up (this may take a minute or 2).
```
```suggestion
for a file `admin_operator.pfx` to show up (this may take a minute or two):
```

MoritzThomasHuebner replied:
Fixed

voisey commented:
Thanks for this! I'll have a look at the changes, but in the meantime, a few points regarding the CI:
1. To make the "Semantic PR" check happy, can you please rename the PR to use "docs" at the start, rather than "fix(docs)"? See the available options [here](https://github.com/magma/magma/blob/master/.github/workflows/semantic-pr.yml#L65)
2. Regarding "Markdown lint check", please have a look at the [docs precommit commands](https://github.com/magma/magma/wiki/Contributing-Documentation#precommit)
3. You'll need to rebase to get the "build-all" jobs working

voisey commented on docs/readmes/basics/prerequisites.md:
```diff
@@ -60,7 +60,7 @@ Development can occur from multiple OS's, where **macOS** and **Ubuntu** are **e
    1. [Docker](https://docs.docker.com/engine/install/ubuntu/) and [Docker Compose](https://docs.docker.com/compose/install/)
    2. [VirtualBox](https://www.virtualbox.org/wiki/Linux_Downloads)
-   3. [Vagrant](https://www.vagrantup.com/downloads)
+   3. [Vagrant](https://www.vagrantup.com/downloads) (Install by downloading the `.deb` file. Installing via the command line using `apt-get` can currently cause an issue with OpenSSL.)
```
What's the issue that gets caused with OpenSSL? Could we potentially link to an external page somewhere that explains the issue so the problem can be tracked?

MoritzThomasHuebner replied:
https://github.com/hashicorp/vagrant/issues/12751

I added this link. They also don't specify what the issue is exactly but they found the same fix.

voisey commented on docs/readmes/basics/prerequisites.md:
````diff
@@ -106,6 +106,8 @@ Development can occur from multiple OS's, where **macOS** and **Ubuntu** are **e
       ```bash
       apt install -y make build-essential libssl-dev zlib1g-dev libbz2-dev    libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev  libncursesw5-dev xz-utils tk-dev libffi-dev liblzma-dev python-openssl git
       ```
+   **Note**: For Ubuntu 22.04 use `python3-openssl` instead of `python-openssl`.
````
```suggestion
   **Note**: For Ubuntu 22.04, use `python3-openssl` instead of `python-openssl`.
```

MoritzThomasHuebner replied:
Fixed

voisey commented on docs/readmes/basics/prerequisites.md:
````diff
@@ -140,14 +145,15 @@ Development can occur from multiple OS's, where **macOS** and **Ubuntu** are **e
 
       ```bash
-      pip3 install ansible fabric3 jsonpickle requests PyYAML
+      pip3 install ansible 'fabric<2.0' jsonpickle requests PyYAML
````
Why do you have to install fabric < 2.0?

MoritzThomasHuebner replied:
Thanks, yes upon revisiting this it appears I misunderstood something about the different fabric versions that exist. I reverted this change.

voisey commented on docs/readmes/basics/prerequisites.md:
````diff
@@ -140,14 +145,15 @@ Development can occur from multiple OS's, where **macOS** and **Ubuntu** are **e
    ```bash
    vagrant plugin install vagrant-vbguest
    ```
+    Make sure `virtualbox` is the default provider for `vagrant` by adding the following line to your `.bashrc` (or equivalent) and restart your shell: `export VAGRANT_DEFAULT_PROVIDER="virtualbox"`
````
```suggestion
    Make sure `virtualbox` is the default provider for `vagrant` by adding the following line to your `.bashrc` (or equivalent) and restarting your shell: `export VAGRANT_DEFAULT_PROVIDER="virtualbox"`.
```

MoritzThomasHuebner replied:
Fixed.

MoritzThomasHuebner committed:
docs: Fixed suggestions made during review

- Reverted instructions to install python 3.10.4. Instead, linked to a fix in case the pyenv installation for 3.7.3 does not work.
- Added a link to a recent issue with Vagrant and OpenSSL.
- Reverted instructions about `fabric` installation
- Fixed a few typos.
- Ran `make precommit_fix` to resolve linting issues

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

Neudrino commented on docs/readmes/basics/quick_start_guide.md:
````diff
@@ -133,15 +133,27 @@ Creating orc8r_controller_1       ... done
+You need to change ownership of these files to your user with `chown`, e.g.
 
+```bash
+HOST [magma/orc8r/cloud/docker] sudo chown username:username ../../../.cache/test_certs/admin_cert.key.pem
+HOST [magma/orc8r/cloud/docker] sudo chown username:username ../../../.cache/test_certs/admin_cert.pfx
````
_optional:_ I see that you aligned with lines 139 and 157, however, it might be simpler an more understandable to write something like:
`HOST [magma] sudo chown username:username .cache/test_certs/admin_cert.pfx`
and also update the above mentioned (so far non changed) lines.

Neudrino commented on docs/readmes/basics/quick_start_guide.md:
````diff
@@ -133,15 +133,27 @@ Creating orc8r_controller_1       ... done
+HOST [magma/orc8r/cloud/docker] sudo chown username:username ../../../.cache/test_certs/admin_cert.pfx
+```
+
+Replace `username` with your username, then:
````
This description could be avoided if we use the env-variable `${USER}` instead of `username` above.
I.e. `HOST [magma] sudo chown ${USER}:${USER} .cache/test_certs/admin_cert.pfx`

voisey replied:
@MoritzThomasHuebner I think this would be a nice change to have. Could you please make the change and then we can merge?

Neudrino commented on docs/readmes/basics/quick_start_guide.md:
```diff
@@ -214,7 +226,8 @@ NMS can take upto 60 seconds to finish starting up.
 so go to [host.localhost](https://host.localhost) and login with the same credentials.
 Once there, you can click on the "Organizations" tab on the left sidebar, choose
-`magma-test`, and then check "Enable all networks" in the subsequent pop-up window.
+`magma-test` (the three dots on the right) &rarr; View &rarr; Edit &rarr; Advanced Settings
```
:rocket: I was not aware that `&rarr;` would display and correctly. But it seems so.
